internal/mode/static/nginx/runtime/verify.go

```diff
@@ -0,0 +1,92 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// verifyClient is a client for verifying the config version.
+type verifyClient struct {
+	client  *http.Client
+	timeout time.Duration
+}
+
+// newVerifyClient returns a new client pointed at the config version socket.
+func newVerifyClient(timeout time.Duration) *verifyClient {
+	return &verifyClient{
+		client: &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+					return net.Dial("unix", "/var/run/nginx/nginx-config-version.sock")
+				},
+			},
+		},
+		timeout: timeout,
+	}
+}
+
+// GetConfigVersion gets the version number that we put in the nginx config to verify that we're using
+// the correct config.
+func (c *verifyClient) GetConfigVersion() (int, error) {
+	ctx := context.Background()
+	reqContext, cancel := context.WithTimeout(ctx, c.timeout)
+	defer cancel()
+	req, err := http.NewRequestWithContext(reqContext, "GET", "http://config-version/configVersion", nil)
```

Comment:
Per an earlier comment, maybe let's have the path be `version`. Also, what determines the domain to be `config-version`?

Reply:
Yeah, that makes sense, I've changed it.

> Also, what determines the domain to be config-version?
We can set it to whatever we want, as it's over a unix socket. If there's an error though, it looks clear what the endpoint is trying to do, e.g.:

```
{"level":"error","ts":"2023-09-06T15:32:47Z","logger":"eventHandler","msg":"Failed to update NGINX configuration","error":"failed to reload NGINX: could not get newest config version: could not get expected version 2: error getting client: Get \"http://config-version/version\": dial unix /var/run/nginx/nginx-config-version.sock: connect: no such file or directory",<...>

{"level":"error","ts":"2023-09-06T15:35:11Z","logger":"eventHandler","msg":"Failed to update NGINX configuration","error":"failed to reload NGINX: could not get newest config version: could not get expected version 2: error getting client: Get \"http://some-thing-else/version\": dial unix /var/run/nginx/nginx-config-version.sock: connect: no such file or directory",<...>
```
(error was forced by removing the entry from the NGINX conf)
I'm happy to change it though if something else would make more sense!

Reply:
Thanks for the info. So since the DialContext of the client is the socket, any request we send using that client, regardless of domain, will go over that socket, correct?

I'm fine with the name.

Reply:
> Thanks for the info. So since the DialContext of the client is the socket, any request we send using that client, regardless of domain, will go over that socket, correct?

That's my understanding, yes.